var/spack/packages/R/package.py

```diff
@@ -1,33 +1,49 @@
 from spack import *
 
+
 class R(Package):
-    """R is 'GNU S', a freely available language and environment for
-       statistical computing and graphics which provides a wide va
-       riety of statistical and graphical techniques: linear and
-       nonlinear modelling, statistical tests, time series analysis,
-       classification, clustering, etc. Please consult the R project
-       homepage for further information."""
-    homepage = "http://www.example.com"
-    url      = "http://cran.cnr.berkeley.edu/src/base/R-3/R-3.1.2.tar.gz"
+    """
+    R is 'GNU S', a freely available language and environment for statistical computing and graphics which provides a
+    wide variety of statistical and graphical techniques: linear and nonlinear modelling, statistical tests, time series
+    analysis, classification, clustering, etc. Please consult the R project homepage for further information.
+    """
+    homepage = "https://www.r-project.org"
+    url = "http://cran.cnr.berkeley.edu/src/base/R-3/R-3.1.2.tar.gz"
 
+    version('3.2.3', '1ba3dac113efab69e706902810cc2970')
+    version('3.2.2', '57cef5c2e210a5454da1979562a10e5b')
+    version('3.2.1', 'c2aac8b40f84e08e7f8c9068de9239a3')
+    version('3.2.0', '66fa17ad457d7e618191aa0f52fc402e')
+    version('3.1.3', '53a85b884925aa6b5811dfc361d73fc4')
     version('3.1.2', '3af29ec06704cbd08d4ba8d69250ae74')
 
-    depends_on("readline")
-    depends_on("ncurses")
-    depends_on("icu")
-    depends_on("glib")
-    depends_on("zlib")
-    depends_on("libtiff")
-    depends_on("jpeg")
-    depends_on("cairo")
-    depends_on("pango")
-    depends_on("freetype")
-    depends_on("tcl")
-    depends_on("tk")
+    variant('external-lapack', default=False, description='Links to externally installed BLAS/LAPACK')
+
+    # Virtual dependencies
+    depends_on('blas', when='+external-lapack')
+    depends_on('lapack', when='+external-lapack')
```

Comment:
> Second, I would need something to pass the +shared or ~shared variants through virtual deps, so that you could depend on blas+shared. Right now I think the lack of that capability complicates R.

For this part, what do you think about adding a new keyword to the `depends_on` directive? My idea would be to add an optional `forward_variants` keyword that can be a sequence or a Boolean. Then : 
- if `True` it will forward all the variant values to the dependency
- if `False` it won't forward anything and will use the dependency default (i.e. the current behavior)
- if it is a sequence it will forward only the  variants specified in the sequence
I would make `forward_variants=False` the default to be backward compatible. The idea is that with this in place we may write something like : 
```python
depends_on('blas', forward_variants=['debug', 'shared'] )
```
If you like the idea I'll start implementing something along this line. Let me know